meson.build

```
@@ -352,12 +336,9 @@ if cc.has_header('malloc.h')
 	 conf.set('HAVE_MALLOC_TRIM', '1')
   endif
 endif
-if cc.has_header('cpuid.h') and cc.has_header_symbol('cpuid.h', '__get_cpuid_count') and (host_cpu == 'x86' or host_cpu == 'x86_64')
+has_cpuid = cc.has_header_symbol('cpuid.h', '__get_cpuid_count', required: get_option('plugin_msr'))
+if has_cpuid
```

```suggestion
if  cc.has_header_symbol('cpuid.h', '__get_cpuid_count', required: get_option('plugin_msr'))
```

I don't think this really needs a temporary variable that is only used once.